scripts/generate_tilesets.py

```diff
@@ -197,22 +197,26 @@ def assert_year(img_src, year, tiles_gdf):
         tiles_gdf (GeoDataframe): tiles geodataframe
     """
 
+
     try:
         assert year=='multi-year' and 'year_tile' in tiles_gdf.keys() or str(year).isnumeric() and 'year_tile' not in tiles_gdf.keys()
     except:
-        if img_src=='XYZ' and year=='multi-year':
+        if year=='multi-year':
             logger.error("Option 'multi-year' chosen but the tile geodataframe does not contain a year column. " 
                         "Please add it or set a numeric year in the configuration file.")
-        elif img_src=='XYZ' and year:
+            sys.exit(1)
+        elif year:
             logger.error("Option 'year' chosen but the tile geodataframe contains a year column. " 
                         "Please delete it or set the 'multi-year' option in the configuration file. ")
-        elif img_src=='XYZ':
-            logger.error("Please provide a year in the configuration file.")
+            sys.exit(1)
         elif 'year_tile' in tiles_gdf.keys():
-            logger.error("Option 'year' not chosen but the tile geodataframe contains a year column. " 
-                        "Please set 'year: multi-year' in the configuration file.")
-        sys.exit(1)
-
+            if img_src=='WMS':
+                logger.error("Connector=WMS, delete the year column in the tile geodataframe")            
```

Comment:
In this case, I don't know if it is necessary to stop the code. Wouldn't it be enough to just warn that the year will not be considered? It would be the same as for the MIL connector and the FOLDER connector only use it for the id decomposition.
I would change the beginning of the message with something like "The connector WMS does not support a specific year, ...".

Reply:
Yeah it was my first intention. 
But here if the script is not exited, it leads to an error afterward because the id column contains the year info as well. It is produced if a year is provided when running `prepare_data.py`. So in this case the tile gdf has to be reprocessed without year.
But we can add what you suggest in case a year argument is set in the config file.

Reply:
I have changed the message according to your suggestion.